File: t/CS/WiaDemo/UI/WiaPropertiesErrorsForm.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Leadtools.Demos;

namespace WiaDemo
{
   public partial class WiaPropertiesErrorsForm : Form
   {
      public WiaPropertiesErrorsForm()
      {
         InitializeComponent();
      }

      private void WiaPropertiesErrorsForm_Load(object sender, EventArgs e)
      {
         _lvErrors.Columns.Add(DemosGlobalization.GetResxString(GetType(), "Resx_DeviceName"), 150, HorizontalAlignment.Left);
         _lvErrors.Columns.Add(DemosGlobalization.GetResxString(GetType(), "Resx_ItemName"), 100, HorizontalAlignment.Left);
         _lvErrors.Columns.Add(DemosGlobalization.GetResxString(GetType(), "Resx_PropertyName"), 120, HorizontalAlignment.Left);
         _lvErrors.Columns.Add(DemosGlobalization.GetResxString(GetType(), "Resx_PropertyValue"), 100, HorizontalAlignment.Left);
         _lvErrors.Columns.Add(DemosGlobalization.GetResxString(GetType(), "Resx_Error"), 70 , HorizontalAlignment.Left);

         foreach (MyPropertiesErrors i in MainForm._errorList)
         {
            ListViewItem item;

            item = _lvErrors.Items.Add(i.DeviceName);

            item.SubItems.Add(i.ItemName);
            item.SubItems.Add(i.PropertyName);
            item.SubItems.Add(i.PropertyValue);
            item.SubItems.Add(i.ErrorCodeString);
         }
      }

      private void _btnClear_Click(object sender, EventArgs e)
      {
         MainForm._errorList.Clear();
         _lvErrors.Items.Clear();
      }
   }
}

File: t/VB/WiaDemo/UI/WiaPropertiesErrorsForm.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Partial Public Class WiaPropertiesErrorsForm : Inherits Form
   Public Sub New()
      InitializeComponent()
   End Sub

   Private Sub WiaPropertiesErrorsForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
      _lvErrors.Columns.Add("Device Name", 150, HorizontalAlignment.Left)
      _lvErrors.Columns.Add("Item Name", 100, HorizontalAlignment.Left)
      _lvErrors.Columns.Add("Property Name", 120, HorizontalAlignment.Left)
      _lvErrors.Columns.Add("Property Value", 100, HorizontalAlignment.Left)
      _lvErrors.Columns.Add("Error", 70, HorizontalAlignment.Left)

      For Each i As MyPropertiesErrors In MainForm._errorList
         Dim item As ListViewItem

         item = _lvErrors.Items.Add(i.DeviceName)

         item.SubItems.Add(i.ItemName)
         item.SubItems.Add(i.PropertyName)
         item.SubItems.Add(i.PropertyValue)
         item.SubItems.Add(i.ErrorCodeString)
      Next i
   End Sub

   Private Sub _btnClear_Click(ByVal sender As Object, ByVal e As EventArgs) Handles _btnClear.Click
      MainForm._errorList.Clear()
      _lvErrors.Items.Clear()
   End Sub
End Class
